library/src/test/java/com/datatorrent/lib/bandwidth/BandwidthManagerTest.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.lib.bandwidth;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.apache.commons.io.FileUtils;
+
+import com.datatorrent.api.Attribute;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DAG;
+import com.datatorrent.lib.helper.OperatorContextTestHelper;
+
+public class BandwidthManagerTest
+{
+  private static class TestMeta extends TestWatcher
+  {
+    private String applicationPath;
+    private BandwidthManager underTest;
+    private Context.OperatorContext context;
+    private long bandwidthLimit = 10L;
+    private ScheduledExecutorTestService mockschedular;
```

Comment:
I guess it will be more readable if we keep it as indipendent class as size of class is pretty big. Also it is accessed from withing private class, so I am moving it at "default" access level in same file.